### PR TITLE
Simplified hash tree root

### PR DESCRIPTION
**Pull request by vbuterin**

### CODE REVIEW

File: test_libs/pyspec/eth2spec/utils/minimal_ssz.py [hunk 1]
```diff
@@ -267,48 +267,35 @@ def infer_type(value):
...
+def hash_tree_root(value, typ=None):
+    if typ is None:
+        typ = infer_type(value)
+    chunks = pack_object(value, typ)
+    if (isinstance(typ, list) and len(typ) == 1) or typ == 'bytes':
+        return mix_in_length(merkleize(chunks), len(value))
+    else:
+        if isinstance(typ, list):
+            assert len(value) == typ[1]
+        elif typ[:5] == 'bytes':
```
Comment (vbuterin) [inline test_libs/pyspec/eth2spec/utils/minimal_ssz.py]:
```suggestion
        elif isinstance(typ, str) and typ[:5] == 'bytes':
```

**Comment by djrtwo** [comment]
This looks good @vbuterin , but should we wait on the work by @protolambda  in #1077 ?

**Comment by protolambda** [comment]
With the new version implemented by @vbuterin in #1077, my typing code, and the change to split into `ssz_impl` (new `minimal_ssz` version) and `ssz_typing` (python typing code), I think this PR is simply not applicable anymore. New version can be found here: https://github.com/ethereum/eth2.0-specs/blob/6f82480df2acd5627fee600aeccf0b975c8a1022/test_libs/pyspec/eth2spec/utils/ssz/ssz_impl.py#L145

Closing this PR. Feel free to re-open if necessary.

**Comment by protolambda** [comment]
Also, about SSZ typing. Ping @vbuterin. Your ssz-partials implementation is residing in the `ssz-partials` branch right now. You can open a PR for it. (See #1077 and/or Python serenity call discussion). Link: https://github.com/ethereum/eth2.0-specs/blob/ssz-partials/test_libs/pyspec/eth2spec/utils/ssz/ssz_partials.py

**Pull request closed without merging**